src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -352,13 +355,73 @@ private TextWriter Out
             }
         }
 
+        private Dictionary<TypeDesc, List<MethodDesc>> _methodLists;
+
         private StringBuilder _statics;
         private StringBuilder _gcStatics;
         private StringBuilder _threadStatics;
         private StringBuilder _gcThreadStatics;
 
         // Base classes and valuetypes has to be emitted before they are used.
-        private HashSet<RegisteredType> _emittedTypes;
+        private HashSet<TypeDesc> _emittedTypes;
+
+        private TypeDesc GetFieldTypeOrPlaceholder(FieldDesc field)
+        {
+            try
+            {
+                return field.FieldType;
+            }
+            catch
+            {
+                // Catch errors due to missing dependencies
+                return _compilation.TypeSystemContext.GetWellKnownType(WellKnownType.Boolean);
+            }
+        }
+
+        private void ExpandTypes()
```

Comment:
Temporary bridging code. It will go away once the data structures are switched over to the dependency analysis framework as well.